gen/templates/resource.go.tmpl

```diff
@@ -485,7 +564,14 @@ func getAndSet{{.ResourceClassName}}Attributes(ctx context.Context, diags *diag.
 					{{- end}}
 				{{- end}}
 			}
-			
+
+
+			{{- range .Properties}}
+			if data.{{.Name}}.IsUnknown() {
+			data.{{.Name}} = types.StringNull()
+			}
+			{{- end}}
+
```

Comment:
Could you check the templates for indentation, makes it a bit easier to read I think, since the templates are already hard to read.
```suggestion
			{{- range .Properties}}
			if data.{{.Name}}.IsUnknown() {
			    data.{{.Name}} = types.StringNull()
			}
			{{- end}}

```